stanford_media.module

```diff
@@ -5,310 +5,83 @@
-use Drupal\Core\Link;
-use Drupal\Core\Url;
-use Drupal\file\Entity\File;
-use Drupal\stanford_media\Plugin\media\Source\Image;
-use Drupal\stanford_media\Plugin\audio_embed_field\Provider\StanfordSoundCloud;
-
-/**
- * Implements hook_ENTITY_TYPE_presave().
- *
- * The uuid of the icon file changes so we want to set it when the embed button
- * is saved.
- */
-function stanford_media_embed_button_presave(EntityInterface $entity) {
-  if ($entity->id() == 'media_browser') {
-    /** @var \Drupal\Core\Entity\EntityRepositoryInterface $repository */
-    $repository = \Drupal::service('entity.repository');
-
-    // File does exist already, so lets bail.
-    if ($entity->icon_uuid && $repository->loadEntityByUuid('file', $entity->icon_uuid)) {
-      return;
-    }
-
-    // Copy the media browser icon over and set it on the embed button.
-    $module_handler = \Drupal::moduleHandler()->getModule('stanford_media');
-    $icon = $module_handler->getPath() . '/img/media.png';
-    /** @var \Drupal\Core\File\FileSystemInterface $file_system */
-    $file_system = \Drupal::service('file_system');
-
-    $directory = 'public://media/image/';
-
-    $file_system->prepareDirectory($directory, FileSystemInterface::CREATE_DIRECTORY);
-    $destination = $file_system->copy($icon, $directory . $file_system->basename($icon), FileSystemInterface::EXISTS_REPLACE);
-
-    if ($destination) {
-      $file_properties = [
-        'uri' => $destination,
-        'uuid' => '3ed72229-7529-4cf5-b679-1d7a86b1030a',
-      ];
-
-      $file = File::create($file_properties);
-      $file->save();
-      $entity->set('icon_uuid', $file->uuid());
-    }
-  }
-}
+use Drupal\editor\Entity\Editor;
+use Drupal\stanford_media\Form\MediaLibraryFileUploadForm;
 
 /**
- * Implements hook_ENTITY_TYPE_presave().
+ * Implements hook_ckeditor_css_alter().
  */
-function stanford_media_view_presave(EntityInterface $entity) {
-  // Disable the core view since we use our own.
-  if ($entity->id() == 'media') {
-    $entity->disable();
-  }
+function stanford_media_ckeditor_css_alter(array &$css, Editor $editor) {
+  $css[] = drupal_get_path('module', 'stanford_media') . '/css/stanford_media.css';
 }
 
 /**
- * Implements hook_form_alter().
+ * Implements hook_theme_registry_alter().
  */
-function stanford_media_form_alter(&$form, FormStateInterface $form_state, $form_id) {
-  if ($form_id == 'media_image_edit_form' && !empty($form_state->getUserInput())) {
-    $form['#attached']['library'][] = 'stanford_media/admin';
-  }
-  if (strpos($form_id, 'entity_browser_') !== FALSE) {
-    $form['selection_display']['#prefix'] = '<div class="selection-action-wrapper">';
-    $form['selection_display']['#suffix'] = '</div>';
-  }
+function stanford_media_theme_registry_alter(&$theme_registry) {
+  // Register the path to the template files.
+  $path = drupal_get_path('module', 'stanford_media') . '/templates';
+  $theme_registry['responsive_image_formatter']['path'] = $path;
```

Comment:
This is probably ok but I would like to think on it a bit longer as I am not sure this module should be the gatekeeper of this template.

Reply:
A theme should still be able to override the template. i'm double checking that now

Reply:
I guess it prevents a theme from overridding it. The alternative is to apply the patch at https://www.drupal.org/project/drupal/issues/3064751 or create its own theme implementation with `hook_theme`

Reply:
Yeah, this function hardwires the path for the template. Going with a different theme hook or suggestion might work.